libraries/TeamSpeak3/Helper/String.php

```diff
@@ -850,6 +850,16 @@ public function __toString()
     return (string) $this->string;
   }
 
+  /**
+   *  Specify data which should be serialized to JSON.
+   * 
+   * @return string
+   */
+  public function jsonSerialize()
+  {
+    return $this->string;
```

Comment:
Hmm, according to docs, it looks like string to encode must be unicode:

> value
> The value being encoded. Can be any type except a resource.
> **All string data must be UTF-8 encoded.**

[[PHP.net - json_encode](http://php.net/manual/en/function.json-encode.php)]

I think replacing with `return $this->toUtf8()` might be more appropriate.
Either that, or we need to force Unicode by throwing exception when string is not UTF-8 encoded.

Not sure if this is the right scope, but might be worth while to add a couple tests regarding expected behavior with/without a Unicode string. Maybe something like:

- 2nd test: check if UTF-8 string is still UTF-8 after encoding.
- 3rd test: check for exception when encoding fails, if possible.
(But these really belong in the scope of testing `toUtf8()` method, not here)

I'll add these in after I merge yours.